cpp/src/Slice/Parser.cpp

```diff
@@ -2503,6 +2503,22 @@ Slice::ClassDef::createDataMember(const string& name, const TypePtr& type, bool
         return nullptr;
     }
 
+    // Check whether enclosing class has the same name.
+    if (name == this->name())
+    {
+        _unit->error("data member `" + name + "' cannot share the same name as its enclosing class");
```

Comment:
I am not fond of "cannot share the same name": cannot have the same name?

Reply:
Alright! It can be done!

Reply:
or maybe

```
        _unit->error("`" + name + "': member names cannot be the same as their enclosing type");
```

Reply:
I prefer the more information dense error messages, even if some of the information is ancillary. What do you think Bernard? Any preference to message format?